workflow/train_test_split.py

```diff
@@ -0,0 +1,167 @@
+from collections.abc import Iterable
+import numpy as np
+import pandas as pd
+import pytest
+
+
+def train_test_split(
+    df,
+    unique_key,
+    test_size=0.2,
+    seed=0,
+    stratify=None
+):
+    if stratify is None:
+        test_df = get_test_split(df, unique_key, test_size, seed)
+    else:
+        test_df = stratified_split(df, unique_key, stratify, test_size, seed)
+
+    train_df = df[~df[unique_key].isin(test_df[unique_key])]
+
+    return train_df, test_df
+
+
+def get_test_split(df, unique_key, test_size, seed=0, resolution=100000):
+    split_unique_key = df[unique_key].apply(
+        lambda x: ((hash(hash(x) + seed) % resolution))
```

Comment:
```suggestion
        lambda x: hash(hash(x) + seed) % resolution
```

Reply:
``` ^ hash("something")```
is to make the output have the right distribution even if x does not
for example `hash(x) == x` for integers would make `hash(hash(x) + 0) == x` and all bets are off :)
better safe then sorry, don't want to get bad results due to some bad splitting :)

Reply:
`^` (xor) preserves entropy which is a good property for hashing
combining multiple elements into the same hash value can be a bit tricky, but we only have one variable element and a constant one.
for multiple elements you have to watch out that for example the hash is not the same when elements are reordered or that "ab" "c" and "a" "bc" arent hashed to the same value :)